docs/node-lifecycle.md

```diff
@@ -0,0 +1,326 @@
+# The life cycle of a Teleport instance
+
+The top-level `teleport` application acts as a scheduler that manages the 
+execution of various registered "services", and an event bus that 
+communicates between them.
+
+The services can run briefly and exit, or live as long as the 
+process, depending on what they need to do. They can also be 
+tagged as "critical" services, meaning that if the service exits
+with error it will bring down the whole process. 
+
+Most of this is encapsulated by the `./lib/service.TeleportProcess` type, and 
+services usually interact with it via the `./lib/service.Supervisor` interface.
+
+## TeleportProcess
+
+`main`
+  - configure & parse command line (`/tool/teleport/common`)
+  - load config file & merge config (`config.Configure`)
+  - handle command routing for subcommands (e.g. re-exec, scp server side, etc)
+`service.Run()`
+`NewTeleport()`
+    - sets up a bunch of services (see `initSSH()` below)
+`TeleportProcess.Start()` (starts supervisor)
+    - kicks off all of the registered services
+wait for signals
+exit, potentially forking replacement depending on signal.
+
+## Reverse Tunnel
+
+A node can be configured to use a tunnel, where a node will initiate a 
+connection with a proxy that the proxy will route commands over. 
+
+## An SSH Node Lifecycle
+
+`initSSH()`: 
+
+RegisterWithAuthServer() starts the `register.node` critical service.
```

Comment:
`registers` the service for later execution. Nothing actually _happens_ until the supervisor is started.